src/Main.elm

```diff
@@ -302,9 +632,155 @@ getWikipediaCategories record =
               Nothing ->
                [li [] [a [][text("Nothing")]]]
               Just categories ->
-                (List.map getCategoryItem categories)
-
--- Add Category Item to List
-getCategoryItem : Category -> Html msg
-getCategoryItem category = 
-        a [href ("https://en.wikipedia.org/" ++ category.link), target "_blank"] [ li [][ text ((category.text)) ]]
\ No newline at end of file
+                (List.map createCategoryItemElement categories)
+
+-- extract Category Item from Category List and create HTML element
+createCategoryItemElement : Category -> Html msg
+createCategoryItemElement category = 
+
+          a [href ("https://en.wikipedia.org/" ++ category.link), target "_blank",
+                      css [
+                  Tw.no_underline,
+                  Tw.text_color Tw.gray_900
+            ]] 
+          [ li [
+          css[ 
+              Tw.block,
+              Tw.text_xl
+            , Tw.font_serif
+            , Tw.bg_color Tw.black
+            , Tw.text_color Tw.pink_400
+            , Tw.rounded
+            , Tw.px_2
+            , Tw.py_4
+            , Tw.my_0_dot_5
+            , Css.hover [ Tw.bg_color Tw.pink_400, Tw.text_color Tw.white ],
+            Tw.mr_1
+            , Css.lastChild
+                [ Tw.mr_0
+                ]
+          ]
+          ][ text ((category.text))]]
+
+-- Create HTML Element for random items
+createRandomItemElement : Maybe WikipediaRecord -> Html Msg
+createRandomItemElement record =
+        div[
+          css[
+            Tw.pr_6,
+            Tw.pl_4,
+            Tw.py_2,
+            Tw.mb_4,
+            Tw.border,
+            Tw.border_color Tw.gray_900, 
+            Tw.rounded_lg,
+            Tw.bg_color Tw.black,
+            Css.hover [ Tw.bg_color Tw.pink_400]
+            ]
+        ][
+          a[
+            href (extractWikipediaURLFromWikipediaRecord record),
+            target "_blank",
+            css [
+                  Tw.no_underline,
+                  Tw.text_color Tw.gray_900
+            ]
+        ][
+          div[
+            css[      
+                Tw.flex,
+                Tw.flex_row,
+                Tw.justify_start
+            ]][
+        div[
+          css[ 
+            Tw.border,
+            Tw.border_color Tw.gray_900, 
+            Tw.rounded,
+            Tw.text_color Tw.zinc_100,
+            Tw.pr_5
+            ]
+        ][
+           img[src (extractWikipediaImageURLFromWikipediaRecord record),
+               width 100,
+               height 100,
+               css [Tw.bg_color Tw.white]][]],
+          div[
+            css[ 
+              Tw.text_color Tw.pink_400, 
+              Tw.font_serif,
+              Css.hover [Tw.text_color Tw.white ]
+              ]
+            ][
+          h2 [] [extractWikipediaTitleFromWikipediaRecord record]
+          ]
+        ] 
+  ]]
+
+-- HTML Element for a 5-Star rating scale
```

Comment:
Wow, epic.

This sort of non-screen-specific component should definitely be in a separate file.

Reply:
Okay, I will move this and similar components to separate files.